server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/SnapshotSortKey.java

```diff
@@ -80,4 +226,29 @@ public static SnapshotSortKey of(String name) {
             default -> throw new IllegalArgumentException("unknown sort key [" + name + "]");
         };
     }
+
+    public record After(String value, String repoName, String snapshotName) implements Writeable {
+
+        After(StreamInput in) throws IOException {
+            this(in.readString(), in.readString(), in.readString());
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(value);
+            out.writeString(repoName);
+            out.writeString(snapshotName);
+        }
+
+        Predicate<SnapshotInfo> longValuePredicate(ToLongFunction<SnapshotInfo> extractor, SortOrder sortOrder) {
+            final var after = Long.parseLong(value);
+            return sortOrder == SortOrder.ASC ? info -> {
```

Comment:
NB previously we captured `snapshotName` and `repoName` as local variables whereas now we just capture the whole `After`. I don't think that should cause any problems.